clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -183,4 +197,50 @@ public synchronized long lastSuccessfulUpdate() {
     public long refreshBackoff() {
         return refreshBackoffMs;
     }
+
+    /**
+     * Set state to indicate if metadata for all topics in Kafka cluster is required or not.
+     * @param needMetadaForAllTopics boolean indicating need for metadata of all topics in cluster.
+     */
+    public void needMetadataForAllTopics(boolean needMetadaForAllTopics) {
+        this.needMetadataForAllTopics = needMetadaForAllTopics;
```

Comment:
Refactoring partitionsFor does seem worthwhile to me since then Metadata is only modified by subscription (and metadata for subscribed topics is all we care about in a steady state). Maybe we can leave this patch as it is and plan to fix it in that JIRA.